src/FakeItEasy/Expressions/ExpressionArgumentConstraintFactory.cs

```diff
@@ -81,7 +81,7 @@ private IArgumentConstraint GetArgumentConstraintFromExpression(Expression expre
             var trappedConstraints = this.argumentConstraintTrapper.TrapConstraints(() =>
             {
                 expressionValue = InvokeExpression(expression);
-            });
+            }) ?? new IArgumentConstraint[0];

```

Comment:
I don't think this should be necessary... I'll try to understand what's going on.